app_flutter/lib/widgets/lattice.dart

```diff
@@ -0,0 +1,876 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' as math;
+
+import 'package:flutter/gestures.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+typedef Painter = void Function(Canvas canvas, Rect rect);
+
+typedef LatticeTapCallback = void Function(Offset offset);
+
+/// A cell in a [LatticeScrollView].
+@immutable
+class LatticeCell extends _LatticeCell {
```

Comment:
I think this file should be split into smaller files, and just import those into this one. Such as lattice_scrollview.dart and lattice_body.dart

Reply:
I considered that but really there's only two files here, LatticeScrollView (about 120 lines) and everything else (about 750 lines), most of which is currently marked private and would have to be made public if we split it. How strongly do you feel about this?

Reply:
Reviewing it again, I agree with you. I wish it was smaller from a maintenance perspective, but this is fine. I just hope it can be moved into the framework :)